bloom-housing/bloom
# feat: uptake of Seeds Toast component

## Issue Overview

This PR addresses #3434
(and builds on/supersedes #3553)


## Description

We had a version of the old UIC AlertBox which offered a "sticky" option that sort of looked like a "toast notification" but it really didn't work quite like how that works in most other design systems and didn't offer automatic stacking of multiple notifications as well as timeouts.

This PR switches to using the Toast component from Seeds, and adds a new `MessageContext` which provides a mechanism to add new toasts to the stack. This doesn't replace all use of AlertBox across the codebase because in other cases those alerts weren't sticky. We should evaluate as part of #3433 how we can update those to Seeds Alert or Message, or Toast if they should in fact be presented like toasts.

This also removes some previous bespoke mechanisms to preserve alert messages across page navigations via `sessionStorage`. We don't need that anymore because just using the Next router + the Toast stack works fine.

## How Can This Be Tested/Reviewed?

It's a little hard to test because toasts require actions of some kind, but you could try the following in public and partners:

* Sign in, sign out
* Create & update users
* Save data (listings and applications on partners)
* Timing out when filling out a common application while signed in
* Any other feature where you feel you'd expect a toast confirmation after performing an action

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
